modules/account-lib/src/coin/sol/transferBuilder.ts

```diff
@@ -1,37 +1,67 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import { BuildTransactionError } from '../baseCoin/errors';
 import { TransactionBuilder } from './transactionBuilder';
 import { Transaction } from './transaction';
-import { Transaction as SolTransaction } from '@solana/web3.js';
+import { isValidAmount, isValidPublicKey } from './utils';
 import { TransactionType } from '../baseCoin';
+import { InstructionBuilderTypes } from './constants';
+import { Memo, Transfer } from './iface';
 
 export class TransferBuilder extends TransactionBuilder {
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
   protected get transactionType(): TransactionType {
-    return TransactionType.WalletInitialization;
+    return TransactionType.Send;
   }
 
-  /** @inheritdoc */
-  protected buildSolTransaction(): SolTransaction {
-    throw new NotImplementedError('buildImplementation not implemented');
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    this.validateMemo(memo);
+    const memoData: Memo = {
+      type: InstructionBuilderTypes.Memo,
+      params: { memo },
+    };
+    this._instructionsData.push(memoData);
+    return this;
   }
 
-  /** @inheritdoc */
-  protected async buildImplementation(): Promise<Transaction> {
-    throw new NotImplementedError('buildImplementation not implemented');
-  }
+  /**
+   *  Set a transfer
+   *
+   * @param {string} fromAddress - the sender address
+   * @param {string} toAddress - the receiver address
+   * @param {string} amount - the amount sent
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  transfer(fromAddress: string, toAddress: string, amount: string): this {
+    if (!fromAddress || !isValidPublicKey(fromAddress)) {
+      throw new BuildTransactionError('Invalid or missing fromAddress, got: ' + fromAddress);
+    }
+    if (!toAddress || !isValidPublicKey(toAddress)) {
+      throw new BuildTransactionError('Invalid or missing toAddress, got: ' + toAddress);
+    }
+    if (!amount || !isValidAmount(amount)) {
+      throw new BuildTransactionError('Invalid or missing amount, got: ' + amount);
+    }
 
-  /** @inheritdoc */
-  protected fromImplementation(rawTransaction: any): Transaction {
-    throw new NotImplementedError('fromImplementation not implemented');
+    const transferData: Transfer = {
+      type: InstructionBuilderTypes.Transfer,
+      params: { fromAddress, toAddress, amount },
+    };
+    this._instructionsData.push(transferData);
```

Comment:
feels like this should be done in a `build()` function.

also, if someone calls `.transfer()` multiple times, this is going to add multiple transfer instructions into this builder. is that intentional?

Reply:
`instructionsData` is just where i put the data used to build the instructions in the build step, and yes, multiple transfer are allowed